packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -326,7 +326,10 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
             // We can't use `npm ci` because some projects don't have npm
             // lockfiles that are up-to-date. Hopefully npm won't decide
             // to change the versions randomly.
-            const install = await execUtils.pipevp(`npm`, [`install`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
+            const install = await execUtils.pipevp(`npm`, [
+              `install`,
+              ...[`dev`, `prod`, `optional`, `peer`].map(include => `--include=${include}`),
```

Comment:
I don't see any mentions of this flag in the 6.x docs so we should filter out `NODE_ENV` instead.

Reply:
👍  done